src/proxy/mod.rs

```diff
@@ -145,18 +145,31 @@ pub async fn copy_hbone(
     let client_to_server = async {
         let mut ri = tokio::io::BufReader::with_capacity(HBONE_BUFFER_SIZE, &mut ri);
         let mut wo = tokio::io::BufWriter::with_capacity(HBONE_BUFFER_SIZE, &mut wo);
-        let res = tokio::io::copy(&mut ri, &mut wo).await;
-        info!(?res, ?desc, "hbone -> tcp");
-        res.expect("");
-        wo.shutdown().await
+        match tokio::io::copy(&mut ri, &mut wo).await {
+            Ok(len) => {
+                info!(len, ?desc, "hbone -> tcp");
+                wo.shutdown().await
+            }
+            Err(e) => {
+                error!("hbone -> tcp: {e}");
+                Err(e)
+            }
+        }
     };
 
     let server_to_client = async {
         let mut ro = tokio::io::BufReader::with_capacity(HBONE_BUFFER_SIZE, &mut ro);
         let mut wi = tokio::io::BufWriter::with_capacity(HBONE_BUFFER_SIZE, &mut wi);
-        let res = tokio::io::copy(&mut ro, &mut wi).await;
-        info!(?res, ?desc, "tcp -> hbone");
-        wi.shutdown().await
+        match tokio::io::copy(&mut ro, &mut wi).await {
+            Ok(len) => {
+                info!(len, ?desc, "tcp -> hbone");
+                wi.shutdown().await
+            }
+            Err(e) => {
+                error!("tcp -> hbone: {e}");
```

Comment:
Why do we need to log in all of these places when we already log at higher levels? Having multiple logs for 1 failure is confusing. Typically should either propogate the error or handle it, not both

Reply:
I kept the 'expect(msg)', will remove it.